packages/cli/openapi-parser/src/schema/examples/getExample.ts

```diff
@@ -1,29 +1,49 @@
 import { OpenAPIV3 } from "openapi-types";
 
-export function getExampleAsNumber(schema: OpenAPIV3.SchemaObject): number | undefined {
+export function getExampleAsNumber(schema: OpenAPIV3.SchemaObject, exampleUnderride?: number): number | undefined {
     if (schema.example != null && typeof schema.example === "number") {
         return schema.example;
     }
-    return undefined;
+    if (exampleUnderride && typeof exampleUnderride !== "number") {
+        // eslint-disable-next-line no-console
+        console.warn(`Expected exampleUnderride to be a number, but got ${typeof exampleUnderride}`);
+        return undefined;
+    }
+    return exampleUnderride;
 }
 
-export function getExampleAsBoolean(schema: OpenAPIV3.SchemaObject): boolean | undefined {
+export function getExampleAsBoolean(schema: OpenAPIV3.SchemaObject, exampleUnderride?: boolean): boolean | undefined {
     if (schema.example != null && typeof schema.example === "boolean") {
         return schema.example;
     }
-    return undefined;
+    if (exampleUnderride && typeof exampleUnderride !== "boolean") {
+        // eslint-disable-next-line no-console
+        console.warn(`Expected exampleUnderride to be a boolean, but got ${typeof exampleUnderride}`);
```

Comment:
Ideally we should pass in the `context` and use the `context.logger` instead of `console.log` directly. That way you can get rid of the eslint disable rule as well